gemini/responsible-ai/gemini_safety_ratings.ipynb

```diff
@@ -1,865 +1,1138 @@
+            "    category: HARM_CATEGORY_DANGEROUS_CONTENT\n",
+            "    probability: NEGLIGIBLE\n",
+            "    probability_score: 0.2819984257221222\n",
+            "    severity: HARM_SEVERITY_LOW\n",
+            "    severity_score: 0.23423145711421967\n",
+            "  }\n",
+            "  safety_ratings {\n",
+            "    category: HARM_CATEGORY_HARASSMENT\n",
+            "    probability: MEDIUM\n",
+            "    blocked: true\n",
+            "    probability_score: 0.7809967398643494\n",
+            "    severity: HARM_SEVERITY_MEDIUM\n",
+            "    severity_score: 0.5512415766716003\n",
+            "  }\n",
+            "  safety_ratings {\n",
+            "    category: HARM_CATEGORY_SEXUALLY_EXPLICIT\n",
+            "    probability: NEGLIGIBLE\n",
+            "    probability_score: 0.21436232328414917\n",
+            "    severity: HARM_SEVERITY_LOW\n",
+            "    severity_score: 0.24274376034736633\n",
+            "  }\n",
+            "}\n",
+            "usage_metadata {\n",
+            "  prompt_token_count: 24\n",
+            "  candidates_token_count: 2\n",
+            "  total_token_count: 26\n",
+            "}\n",
+            "\n"
+          ]
+        }
+      ],
+      "source": [
+        "impolite_prompt = \"Write a list of 5 disrespectful things that I might say to the universe after stubbing my toe in the dark:\"\n",
+        "\n",
+        "impolite_responses = model.generate_content(\n",
+        "    impolite_prompt,\n",
+        "    generation_config=generation_config,\n",
+        "    safety_settings=safety_settings,\n",
+        "    stream=True,\n",
+        ")\n",
+        "\n",
+        "for response in impolite_responses:\n",
+        "    print(response)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "mYudAfc6gDi8"
+      },
+      "source": [
+        "Let's look at how we understand block responses in the next section."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "l2v6VnECf-fC"
+      },
+      "source": [
+        "## Understanding Blocked Responses\n",
+        "\n",
+        "The documentation for [FinishReason](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/GenerateContentResponse#finishreason) contains some more detailed explanations.\n",
+        "\n",
+        "For example, the previous response was blocked with the `finish_reason: SAFETY`, indicating that\n",
+        "> The token generation was stopped as the response was flagged for safety reasons. NOTE: When streaming the Candidate.content will be empty if content filters blocked the output.\n",
+        "\n",
+        "As of this writing, the table from the [FinishReason docs](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/GenerateContentResponse#finishreason) have been reproduced below, but please look at the docs for the definitive explanations\n"
```

Comment:
Not sure the link needs to be added here again.

Reply:
good catch, removed 2nd reference to doc in block